### PR TITLE
Feature/#77 create sms and email template infrastructure

### PR DESCRIPTION
**Pull request by botirjonabdumannonovv**
Closes #250

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/SmsTemplateService.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var validationResult = validator.Validate(smsTemplate);
+        
+        if (!validationResult.IsValid)
+            throw new ValidationException(validationResult.Errors);
+
+        return smsTemplateRepository.CreateAsync(smsTemplate, saveChanges, cancellationToken);
+    }
+
```
Comment (Xolmuratov02) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/SmsTemplateService.cs:41]: Remove unnecessary row

**Review by Xolmuratov02 — APPROVED** [review]
Honey code

**Comment by botirjonabdumannonovv** [comment]
> Honey code

Honey code bro)

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/SmsTemplateService.cs (hunk 1 above)
Comment (botirjonabdumannonovv) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/SmsTemplateService.cs:41]: Understand)

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Controllers/NotificationsController.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+    
+    [HttpGet("templates/email")]
+    public async ValueTask<IActionResult> GetEmailTemplates([FromQuery] FilterPagination filterPagination,
+        CancellationToken cancellationToken)
+    {
+        var result = await emailTemplateService.GetAsync(filterPagination.ToQueryPagination(true).ToQuerySpecification(),
+            cancellationToken);
+
+        return result.Any() ? Ok(mapper.Map<SmsTemplateDto>(result)) : NotFound();
```
Comment (Habiba67y) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Controllers/NotificationsController.cs:30]: mappingda emailTemplateDto ga map qilish kerakmasmi

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/EmailTemplate.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using Type = AirBnB.Domain.Enums.NotificationType;
+using AirBnB.Domain.Common;
+
+namespace AirBnB.Domain.Entities;
+
+/// <summary>
+/// Represents an email template for notifications, implementing the IAuditableEntity interface.
+/// </summary>
+public class EmailTemplate : NotificationTemplate, IAuditableEntity
```
Comment (Habiba67y) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/EmailTemplate.cs:9]: IAuditableEntity dan inherit olish kerakmas

**Review by bunyodkamalov — APPROVED** [review]

**Review by sulton-max — DISMISSED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]

**Pull request merged**